### DISCUSSION TITLE
Getting item info on hover or any other way

### DISCUSSION THREAD

**Discussion by Ingeniouz** [peter-kish/gloot#69] [Q&A] [answered]
Hello, I want to make random attributes on items with the same id, so I need to display these attributes somehow. Can I make an item info panel on hovering an item or something like that? Or is the only way to know the item is by it's image?
P.S. I use InventoryGrid, but can choose another inventory type if it fits the goal better.

**Comment by peter-kish** [comment]
I suppose the problem here is knowing which item is under the mouse cursor?

Unfortunately, I see only one way of doing this at the current state of the plugin: Getting the field coordinates with `CtrlInventoryGrid.get_field_coords` and then looping through the items to see which item is placed on the given coordinates.

A far better way would be to have `CtrlInventoryGrid` signals similar to the `mouse_entered()` and `mouse_exited()` signals of the `Control` class. For example `mouse_entered_item(item)` and `mouse_exited_item(item)` signals which would triggered when the user hovers over the given items.
Another way would be to define a `CtrlInventoryGrid` method that would just return the item under the mouse cursor (called `get_hovered_item()` or similar)

I will look into implementing one of the above (or both) in the next version.

EDIT: Opened an issue for this: https://github.com/peter-kish/gloot/issues/70

**Reply by Ingeniouz** [reply]
Yeah, it's a long-term question, it's not urgent, just wondering if it's possible or will be. I will wait then, thanks. How can I know when the function is there?

**Reply by peter-kish** [reply]
I will close the related issue (https://github.com/peter-kish/gloot/issues/70) and also write in this thread.
I don't think it's a difficult thing to implement, so I'll try to get it done this weekend or perhaps the next one.

**Reply by Ingeniouz** [reply]
Got it. Thank you!

**Comment by peter-kish** [comment, marked as answer]
`item_mouse_entered(item)` and `item_mouse_exited(item)` have been added to `CtrlInventoryGrid` in https://github.com/peter-kish/gloot/pull/71.

**Reply by Ingeniouz** [reply]
Great news! Thank you! I will try it as soon as I get to Godot.

**Reply by Ingeniouz** [reply]
I tried, it works! Yay!
Also I noticed that the new documentation part has item_mouse_entered() for mouse_leaves too, seems like an error.
https://github.com/peter-kish/gloot/blob/master/docs/ctrl_inventory_grid.md

**Reply by peter-kish** [reply]
Yup, my mistake. I'll fix it shortly. Thanks!
